File: src/main/java/com/cuupa/classificator/services/kb/SemanticResult.java
package com.cuupa.classificator.services.kb;

import com.cuupa.classificator.services.kb.semantic.Metadata;

import java.util.List;

public class SemanticResult {

    private final String topicName;

    private String sender;

    private List<Metadata> metaData;

    public SemanticResult(String topicName, List<Metadata> metaData) {
        this.topicName = topicName;
        this.metaData = metaData;
    }

    public SemanticResult(String topicName) {
        this.topicName = topicName;
    }

    public String getTopicName() {
        return topicName;
    }

    public List<Metadata> getMetaData() {
        return metaData;
    }

    public String getSender() {
        return sender;
    }

    public void setSender(String sender) {
        this.sender = sender;
    }
}

File: src/main/java/com/cuupa/classificator/services/kb/SemanticResult.kt
package com.cuupa.classificator.services.kb

data class SemanticResult(val topicName: String, var sender: String, var metaData: List<Metadata>) {

    constructor(topicName: String, metaData: MutableList<Metadata>) : this(topicName, Sender.UNKNOWN, metaData)

    constructor(topicName: String) : this(topicName, Sender.UNKNOWN, mutableListOf())
}